sphinx/pycode/parser.py

```diff
@@ -525,7 +525,21 @@ def finalize_block(self) -> None:
         definition = self.indents.pop()
         if definition[0] != 'other':
             typ, funcname, start_pos = definition
-            end_pos = self.current.end[0] - 1
+            end_pos, end_col = self.current.end
+
+            def is_implicit_dedent():
+                if end_pos == len(self.buffers) + 1 and end_col == 0:
+                    return True
+                if end_col < len(self.get_line(end_pos)):
+                    return True
```

Comment:
The first of these two conditional `if` blocks handles implicit dedent tokens that occur when the end-of-file is reached (because any open code blocks should be closed at that point).

I'm less confident in explaining what the second conditional is intended to handle - and it may be possible to simplify the logic used for it.